### ISSUE TITLE
Private variable in control structure "for"

### ISSUE THREAD

**Issue by maximilionus** [SkaceKamen/vscode-sqflint#58]
In control structure [`for [{},{},{}] do {}`](https://community.bistudio.com/wiki/for) - variables, defined in this loop like [private](https://community.bistudio.com/wiki/private):
```sqf
for [{ private _i = 0 }, { _i <= 10 }, { _i = _i + 1 }] do {};
```
Shows up as *possibly undefined*
![image](https://user-images.githubusercontent.com/29492157/67030565-563f2a00-f118-11e9-9f9c-272888f070cd.png)

But variable `_i` is defined inside this loop:
![image](https://user-images.githubusercontent.com/29492157/67031292-d619c400-f119-11e9-8a3d-210d0bba0608.png)

**Comment by SkaceKamen** [comment]
Oh man, didn't know this is possible ... so variables defined as private in a block are private for that block, with exception of for loop, where they're private for the loop scope? Damn 😄

**Comment by maximilionus** [comment]
@SkaceKamen exactly! Variables, defined like this, will be local to this loop and will not be overridden at each iteration.
